source/max-staleness/max-staleness.md

```diff
@@ -0,0 +1,520 @@
+
+Consider a primary P and a secondary S, and a client with heartbeatFrequencyMS set to 10 seconds. Say that the primary's
+clock is 50 seconds skewed ahead of the client's.
+
+The client checks P and S at time 60 (meaning 60 seconds past midnight) by the client's clock. The primary reports its
+lastWriteDate is 10.
+
+Then, S reports its lastWriteDate is 0. The client estimates S's staleness as:
+
+```
+(S.lastUpdateTime - S.lastWriteDate) - (P.lastUpdateTime - P.lastWriteDate) + heartbeatFrequencyMS
+= (60 - 0) - (60 - 10) + 10
+= 20 seconds
+```
+
+(Values converted from milliseconds to seconds for the sake of discussion.)
+
+Note that the secondary appears only 10 seconds stale at this moment, but the client adds heartbeatFrequencyMS,
+pessimistically assuming that the secondary will not replicate at all between now and the next check. If the current
+staleness plus heartbeatFrequencyMS is still less than maxStalenessSeconds, then we can safely read from the secondary
+from now until the next check.
+
+The client re-checks P and S 10 seconds later, at time 70 by the client's clock. S responds first with a lastWriteDate
+of 5: it has fallen 5 seconds further behind. The client updates S's lastWriteDate and lastUpdateTime. The client now
+estimates S's staleness as:
+
+```
+(S.lastUpdateTime - S.lastWriteDate) - (P.lastUpdateTime - P.lastWriteDate) + heartbeatFrequencyMS
+= (70 - 5) - (60 - 10) + 10
+= 25 seconds
+```
+
+Say that P's response arrives 10 seconds later, at client time 80, and reports its lastWriteDate is 30. S's staleness is
+still 25 seconds:
+
+```
+(S.lastUpdateTime - S.lastWriteDate) - (P.lastUpdateTime - P.lastWriteDate) + heartbeatFrequencyMS
+= (70 - 5) - (80 - 30) + 10
+= 25 seconds
+```
+
+The same story as a table:
+
+| Client clock | Primary clock | Event           | S.lastUpdateTime | S.lastWriteDate | P.lastUpdateTime | P.lastWriteDate | S staleness |
+| ------------ | ------------- | --------------- | ---------------- | --------------- | ---------------- | --------------- | ----------- |
+| 60           | 10            | P and S respond | 60               | 0               | 60               | 10              | 20 seconds  |
+| 70           | 20            | S responds      | 70               | 5               | 60               | 10              | 25 seconds  |
+| 80           | 30            | P responds      | 70               | 5               | 80               | 30              | 25 seconds  |
+
+> from table import Table
+>
+> data = \[\
+> \['Client clock', 'Primary clock', 'Event', 'S.lastUpdateTime', 'S.lastWriteDate',\
+> 'P.lastUpdateTime',
+> 'P.lastWriteDate', 'S staleness'\],
+>
+> \['60', '10', 'P and S respond', '60', '0', '60', '10', '20 seconds'\], \['70', '20', 'S responds', '70', '5', '60',
+> '10', '25 seconds'\], \['80', '30', 'P responds', '70', '5', '80', '30', '25 seconds'\]
+>
+> \]
+>
+> print Table(data).create_table()
+
```

Comment:
This was a comment in the original code, are we intentionally keep it as a quote?

https://github.com/mongodb/specifications/blob/39add29e0a3b6721688845874383878b90811a12/source/max-staleness/max-staleness.rst?plain=1#L271-L283

```suggestion
```